src/aead/aead.rs

```diff
@@ -613,4 +633,46 @@ mod tests {
                                         &mut in_out, &ad).is_err());
         }
     }
+
+    fn test_aead_inner(aead_alg: &'static aead::Algorithm,
+                       key_bytes: &[u8],
+                       nonce: &[u8],
+                       plaintext: &Vec<u8>,
+                       ad: &[u8],
+                       ct: &[u8],
+                       error: &Option<String>,
+                       in_prefix: &[u8]) {
+        let max_overhead_len = aead_alg.max_overhead_len();
+        let mut s_in_out = plaintext.clone();
+        for _ in 0..max_overhead_len {
```

Comment:
Similarly, maybe we need to optimize this so the tests run faster. Maybe the loop can be replaced with `s_in_out.resize(plantext.len() + max_overhead_len)` or similar?